src/windows/webview.c

```diff
@@ -477,47 +488,107 @@ static ICoreWebView2EnvironmentOptionsVtbl VTBL5 = {
-	WCHAR *dll = NULL;
-	HMODULE lib = NULL;
+	bool ok = false;
+	DWORD flags = KEY_READ;
+
+	HKEY hkey = as_user ? HKEY_CURRENT_USER : HKEY_LOCAL_MACHINE;
+	if (!as_user)
+		flags |= KEY_WOW64_32KEY;
 
-	LSTATUS r = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
-		L"Software\\Microsoft\\EdgeUpdate\\ClientState\\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}",
-		0, KEY_WOW64_32KEY | KEY_READ, &key);
+	wchar_t reg_path[MAX_PATH] = {0};
+	_snwprintf_s(reg_path, MAX_PATH, _TRUNCATE, WEBVIEW_REG_PATH, L"ClientState");
+
+	HKEY key = NULL;
+	LSTATUS r = RegOpenKeyEx(hkey, reg_path, 0, flags, &key);
 	if (r != ERROR_SUCCESS)
 		goto except;
 
-	dll = MTY_Alloc(MAX_PATH, sizeof(WCHAR));
-	DWORD size = MAX_PATH * sizeof(WCHAR);
-
+	DWORD size = MAX_PATH * sizeof(wchar_t);
+	wchar_t dll[MAX_PATH] = {0};
 	r = RegQueryValueEx(key, L"EBWebView", 0, NULL, (BYTE *) dll, &size);
 	if (r != ERROR_SUCCESS)
 		goto except;
 
-	#if defined(_WIN64)
-		const WCHAR *path = L"\\EBWebView\\x64\\EmbeddedBrowserWebView.dll";
+	if (pathw)
+		_snwprintf_s(pathw, MTY_PATH_MAX, _TRUNCATE, L"%s\\%s", dll, WEBVIEW_DLL_PATH);
+
+	ok = true;
+
+	except:
+
+	if (key)
+		RegCloseKey(key);
+
+	return ok;
+}
+
+static bool webview_dll_path_client(wchar_t *pathw, bool as_user)
+{
+	bool ok = false;
+	DWORD flags = KEY_READ;
+
+	HKEY hkey = as_user ? HKEY_CURRENT_USER : HKEY_LOCAL_MACHINE;
+	if (!as_user)
+		flags |= KEY_WOW64_32KEY;
 
-	#else
-		const WCHAR *path = L"\\EBWebView\\x86\\EmbeddedBrowserWebView.dll";
-	#endif
+	wchar_t reg_path[MAX_PATH] = {0};
+	_snwprintf_s(reg_path, MAX_PATH, _TRUNCATE, WEBVIEW_REG_PATH, L"Clients");
 
-	if (wcscat_s(dll, MAX_PATH, path) != 0)
+	HKEY key = NULL;
+	LSTATUS r = RegOpenKeyEx(hkey, reg_path, 0, flags, &key);
+	if (r != ERROR_SUCCESS)
 		goto except;
 
-	lib = LoadLibrary(dll);
-	if (!lib)
+	DWORD size = MAX_PATH * sizeof(wchar_t);
+	wchar_t dll[MAX_PATH] = {0};
+	r = RegQueryValueEx(key, L"location", 0, NULL, (BYTE *) dll, &size);
+	if (r != ERROR_SUCCESS)
 		goto except;
 
-	except:
+	size = MAX_PATH * sizeof(wchar_t);
+	wchar_t version[MAX_PATH] = {0};
+	r = RegQueryValueEx(key, L"pv", 0, NULL, (BYTE *) version, &size);
+	if (r != ERROR_SUCCESS)
+		goto except;
+
+	if (pathw)
+		_snwprintf_s(pathw, MTY_PATH_MAX, _TRUNCATE, L"%s\\%s\\%s", dll, version, WEBVIEW_DLL_PATH);
 
-	MTY_Free(dll);
+	ok = true;
+
+	except:
 
 	if (key)
 		RegCloseKey(key);
 
-	return lib;
+	return ok;
+}
+
+static bool webview_dll_path(wchar_t *pathw, bool as_user)
+{
+	// Try convenient ClientState first
+	if (webview_dll_path_clientstate(pathw, as_user))
+		return true;
+
+	// Construct the base path manually if ClientState is not available
+	return webview_dll_path_client(pathw, as_user);
+}
+
+static HMODULE webview_load_dll(void)
+{
+	wchar_t path[MTY_PATH_MAX] = {0};
+
+	// Try system WebView
+	if (webview_dll_path(path, false))
+		return LoadLibrary(path);
+
+	// Try user WebView
+	if (webview_dll_path(path, true))
+		return LoadLibrary(path);
+
+	return NULL;
 }
```

Comment:
```suggestion
static HMODULE webview_load_dll(void)
{
	wchar_t path[MTY_PATH_MAX] = {0};
	HMODULE ret = NULL;

	// Try system WebView
	if (webview_dll_path(path, false))
		ret = LoadLibrary(path);

	// Try user WebView
	if (!ret && webview_dll_path(path, true))
		ret = LoadLibrary(path);

	return ret;
}
```

Actually try the user one, if the system one doesn't work.

Reply:
Ah, if it doesn't load even with the path, yeah